test/e2e/build_test.go

```diff
@@ -267,6 +267,23 @@ var _ = Describe("Podman build", func() {
 		Expect(session.OutputToString()).To(ContainSubstring("hello"))
 	})
 
+	It("podman build test volume mount on server end", func() {
```

Comment:
This passes on main which means you are not testing the bug!

The issue #17139 shows that you need to have client and server on different machines. You can never test it like this. You need to have a client where the given -v path does not exist and it only exist on the serve side.
You can add a test like this in pkg/machine/e2e as this runs on actually runs two different systems.

Reply:
Yes adding a mount on machine side and then using it in build SGTM, let me amend this.